site/src/Librecores/ProjectRepoBundle/Util/GitHubUtil.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace Librecores\ProjectRepoBundle\Util;
+
+/**
+ * Helper utility service, which provides methods for operating with GitHub projects.
+ */
+
+class GitHubUtil
+{
+
+    private $githubUrl;
+    private $shieldsUrl;
+    private $shieldsExt;
+    private $sourceRepoUrl;
+    private $issueTrackerUrl;
+
+    public function __construct($githubUrl, $shieldsUrl, $shieldsExt)
+    {
+        $this->githubUrl       = $githubUrl;
+        $this->shieldsUrl      = $shieldsUrl;
+        $this->shieldsExt      = $shieldsExt;
+        $this->sourceRepoUrl   = null;
+        $this->issueTrackerUrl = null;
```

Comment:
IMHO this is a bad approach. Once we add more GitHub descriptors (e.g. for releases or Wiki), we will need to add more URLs to this class. It makes the logic a bit less flexible then it was in the original approach, when the class was being constructed once for a single URL